update-dependencies/Program.cs

```diff
@@ -3,152 +3,138 @@
                     {
-                        CreatePullRequest(updateResults).Wait();
+                        Trace.TraceInformation($"Changes made but no GitHub credentials specified, skipping PR creation");
+                    }
+                    else
+                    {
+                        await CreatePullRequest(updateResults);
                     }
                 }
             }
             catch (Exception e)
             {
                 Console.Error.WriteLine($"Failed to update dependencies:{Environment.NewLine}{e.ToString()}");
+                Environment.Exit(1);
             }
-        }
 
-        private static bool ParseArgs(string[] args)
-        {
-            foreach (string arg in args)
-            {
-                if (string.Equals(arg, "--Update", StringComparison.OrdinalIgnoreCase))
-                {
-                    s_updateOnly = true;
-                }
-                else if (arg.Contains('='))
-                {
-                    int delimiterIndex = arg.IndexOf('=');
-                    string name = arg.Substring(0, delimiterIndex);
-                    string value = arg.Substring(delimiterIndex + 1);
-
-                    Environment.SetEnvironmentVariable(name, value);
-                }
-                else
-                {
-                    Console.Error.WriteLine($"Unrecognized argument '{arg}'");
-                    return false;
-                }
-            }
-
-            return true;
+            Environment.Exit(0);
         }
 
-        private static DependencyUpdateResults UpdateFiles()
+        private static async Task<DependencyUpdateResults> UpdateFiles()
         {
-            // Ideally this logic would depend on the CLI produces and consumes metadata.  Since it doesn't
-            // exist various version information is inspected to obtain the latest CLI version along with
-            // the runtime (e.g. shared framework) it depends on.
-
-            BuildInfo cliBuildInfo = BuildInfo.Get(CliBuildInfoName, s_config.CliVersionUrl, fetchLatestReleaseFile: false);
-            // Adjust the LatestReleaseVersion since it is not the full version and all consumers here need it to be.
-            cliBuildInfo.LatestReleaseVersion = $"{s_config.RuntimeReleasePrefix}-{cliBuildInfo.LatestReleaseVersion}";
-            CliDependencyHelper cliDependencyHelper = new CliDependencyHelper(cliBuildInfo.LatestReleaseVersion);
-            string sharedFrameworkVersion = cliDependencyHelper.GetSharedFrameworkVersion();
-
-            IEnumerable<DependencyBuildInfo> buildInfos = new[]
+            Trace.TraceInformation($"Retrieving build info from '{Options.BuildInfoUrl}'");
+            Stream stream = await (new HttpClient().GetStreamAsync(Options.BuildInfoUrl));
+            XDocument buildInfoXml = XDocument.Load(stream);
+            OrchestratedBuildModel buildInfo = OrchestratedBuildModel.Parse(buildInfoXml.Root);
```

Comment:
I was going to ask if `BuildManifestClient.FetchManifestAsync` worked here, but it uses the GitHub API (rather than the `raw` endpoint) so this code is probably worth it to get around the pretty small anonymous rate limit. Might be worth adding functionality to the VersionTools GitHub client itself to derive a `raw` url when fetching something anonymously, actually.